aries_vcx/src/protocols/oob/mod.rs

```diff
@@ -0,0 +1,71 @@
+use crate::error::{VcxError, VcxErrorKind, VcxResult};
+use crate::handlers::out_of_band::OutOfBandInvitation;
+use crate::messages::out_of_band::handshake_reuse::OutOfBandHandshakeReuse;
+use crate::messages::out_of_band::handshake_reuse_accepted::OutOfBandHandshakeReuseAccepted;
+
+pub fn build_handshake_reuse_msg(oob_invitation: &OutOfBandInvitation) -> OutOfBandHandshakeReuse {
+    OutOfBandHandshakeReuse::default()
+        .set_out_time()
+        .set_thread_id_matching_id()
+        .set_parent_thread_id(&oob_invitation.id.0)
+}
+
+pub fn build_handshake_reuse_accepted_msg(
+    handshake_reuse: &OutOfBandHandshakeReuse,
+) -> VcxResult<OutOfBandHandshakeReuseAccepted> {
+    let thread_id = handshake_reuse.get_thread_id();
+    let pthread_id = handshake_reuse.thread.pthid.as_deref().ok_or(VcxError::from_msg(
+        VcxErrorKind::InvalidOption,
+        "Parent thread id missing",
+    ))?;
+    Ok(OutOfBandHandshakeReuseAccepted::default()
+        .set_out_time()
+        .set_thread_id(&thread_id)
+        .set_parent_thread_id(pthread_id))
+}
+
+#[cfg(test)]
+#[cfg(feature = "general_test")]
+mod unit_tests {
+    use crate::handlers::out_of_band::OutOfBandInvitation;
+    use crate::messages::a2a::MessageId;
+    use crate::protocols::oob::{build_handshake_reuse_accepted_msg, build_handshake_reuse_msg};
+    use crate::utils::devsetup::{was_in_past, SetupMocks};
+
+    #[test]
+    #[cfg(feature = "general_test")]
+    fn test_build_handshake_reuse_msg() {
+        let _setup = SetupMocks::init();
+        let mut msg_invitation = OutOfBandInvitation::default();
+        msg_invitation.id = MessageId("invitation-id".to_string());
+        let msg_reuse = build_handshake_reuse_msg(&msg_invitation);
+
+        assert_eq!(msg_reuse.id, MessageId("testid".into()));
+        assert_eq!(msg_reuse.id, MessageId(msg_reuse.thread.thid.unwrap()));
+        assert_eq!(msg_reuse.thread.pthid.unwrap(), "invitation-id");
+        assert!(was_in_past(
+            &msg_reuse.timing.unwrap().out_time.unwrap(),
+            chrono::Duration::milliseconds(100)
+        )
+        .unwrap());
```

Comment:
Also, I very much dislike the copy-pasting in the tests. Think about the reader, who, looking for a documentation of how the codebase works, has to decipher one line difference between copy pasted walls of text.

Reply:
I don't see much shared between the two tests. Perhaps 2-3 setup lines could be extracted, I don't think it's worth it here